bigbluebutton-html5/imports/ui/components/upload/service.js

```diff
@@ -0,0 +1,137 @@
+import { defineMessages, injectIntl } from 'react-intl';
+import axios from 'axios';
+import { UploadRequest } from '/imports/api/upload';
+import Auth from '/imports/ui/services/auth';
+import { notify } from '/imports/ui/services/notification';
+import { makeCall } from '/imports/ui/services/api';
+
+const UPLOAD = Meteor.settings.public.upload;
+const DOWNLOAD = Meteor.settings.public.download;
+
+const intlMessages = defineMessages({
+  uploading: {
+    id: 'app.upload.toast.uploading',
+    description: 'Upload toast file uploading',
+  },
+  completed: {
+    id: 'app.upload.toast.completed',
+    description: 'Upload toast file completed',
+  },
+  401: {
+    id: 'app.upload.toast.error.401',
+    description: 'Upload toast error unauthorized',
+  },
+  408: {
+    id: 'app.upload.toast.error.408',
+    description: 'Upload toast error request timeout',
+  },
+  header: {
+    id: 'app.upload.notification.header',
+    description: 'Uploaded file notification header',
+  },
+  disclaimer: {
+    id: 'app.upload.notification.disclaimer',
+    description: 'Uploaded file notification disclaimer',
+  },
+});
+
+const requestUpload = (source, filename) => {
+  return new Promise((resolve, reject) => {
+    const timestamp = new Date().getTime();
+    makeCall('requestUpload', source, filename, timestamp);
+
```

Comment:
In my local implementation to 2.3a2 server, this gives an error in import/api/upload/server/methods/requestUpload.js (credentials is undefined). 
Changing it to "makeCall('requestUpload', Auth.credentials, source, filename, timestamp);" suppresses the complaint. Does it make sense?